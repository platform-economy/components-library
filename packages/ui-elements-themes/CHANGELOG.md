# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
