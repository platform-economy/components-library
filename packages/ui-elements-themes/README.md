# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
