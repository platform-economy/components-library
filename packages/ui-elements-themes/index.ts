export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
