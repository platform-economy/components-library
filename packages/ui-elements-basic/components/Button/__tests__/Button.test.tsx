import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import enzyme, { mount } from 'enzyme';

import 'jest-styled-components';

import { withTestTheme } from '@relayr/ui-elements-themes';

enzyme.configure({ adapter: new Adapter() });

// Component
import { Button } from '../index';

describe('ui-elements-basic', () => {
  describe('Components', () => {
    describe('Button', () => {
      it('Should render with default props only', () => {
        const tree = TestRenderer
          .create(withTestTheme(<Button.Raw />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('Should render with icons and text', () => {
        const tree = TestRenderer.create(withTestTheme(
          <Button.Raw
            iconLeft="iconLeft"
            iconRight="iconRight"
          >
            Button Primary Text
          </Button.Raw>))
        .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('Should handle onClick event correctly', () => {
        const handler = jest.fn();
        const tree = mount(withTestTheme(<Button.Raw onClick={handler} />));

        tree.find('button').simulate('click');
        expect(handler).toHaveBeenCalledTimes(1);
      });

      it('Should handle onClick event correctly without handler', () => {
        const instance = mount(withTestTheme(<Button.Raw />));
        let error;

        try {
          instance.find('button').simulate('click');
        } catch (err) {
          error = err;
        }
        expect(error).toBe(undefined);
      });
    });
  });
});
