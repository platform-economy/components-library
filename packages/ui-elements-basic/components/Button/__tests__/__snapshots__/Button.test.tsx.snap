// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-basic Components Button Should render with default props only 1`] = `
.c0 {
  font-size: .75em;
  font-weight: bold;
  outline: none;
  text-align: center;
  vertical-align: middle;
  min-height: 35px;
  border: none;
  border-radius: 3px;
  padding: 5px 30px;
  -webkit-transition: 100ms all ease-in-out;
  transition: 100ms all ease-in-out;
  white-space: nowrap;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0 .icon-left {
  margin: 0 5px 0 0;
}

.c0 .icon-right {
  margin: 0 0 0 5px;
}

<button
  className="c0"
/>
`;

exports[`ui-elements-basic Components Button Should render with icons and text 1`] = `
.c0 {
  font-size: .75em;
  font-weight: bold;
  outline: none;
  text-align: center;
  vertical-align: middle;
  min-height: 35px;
  border: none;
  border-radius: 3px;
  padding: 5px 30px;
  -webkit-transition: 100ms all ease-in-out;
  transition: 100ms all ease-in-out;
  white-space: nowrap;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0 .icon-left {
  margin: 0 5px 0 0;
}

.c0 .icon-right {
  margin: 0 0 0 5px;
}

<button
  className="c0"
>
  <span
    className="icon-left"
  >
    iconLeft
  </span>
  Button Primary Text
  <span
    className="icon-right"
  >
    iconRight
  </span>
</button>
`;
